### DISCUSSION TITLE
Why `auto_enclose` ignore `auto_pairs`?

### DISCUSSION THREAD

**Discussion by lua-rocks** [orbitalquark/textadept#126] [Q&A] [answered]
I dumped my `textadept.editing.auto_pairs` variable, it looks like this:

```lua
{
  [34] = '"',
  [39] = "'",
  [40] = ")",
  [91] = "]",
  [123] = "}"
}
```

It is default value and it suits me fine.

Then I enable `auto_enclose`:

```lua
textadept.editing.auto_enclose = true
```

I expected that when I select text and press one of the buttons listed in `textadept.editing.auto_pairs`, it will be enclosed from both sides and it does, but also auto_enclose works with any shift+number keys. For example with `@`, `#`, `$`, etc. How can I force it to process only the specified characters?

**Comment by orbitalquark** [comment, marked as answer]
You cannot change the behavior other than turning `auto_enclose` off and creating your own similar event handler that does what you want.

I think the original desire for `auto_enclose` was to enclose text in any punctuation (e.g. markdown modifiers), not limited to auto-enclose characters.

**Reply by lua-rocks** [reply]
Ok. Thank you for the clarification!

**Reply by metaleap** [reply]
I too would rather wish for a behavior as outlined by @lua-rocks , as I'm used to that for years from all sorts of previous editors I lived in... and Textadept API docs say this:

> `textadept.editing.auto_enclose (bool)`
> Whether or not to auto-enclose selected text when typing a punctuation character, **taking** `textadept.editing.auto_pairs` **into account**

- Couldn't this be made as (A) at-least-hinted (arguably/presumably) in docs and (B) common elsewhere and (C) more useful for the vast masses of non-Markdown-coders out there, who when marking-text-then-typing-something usually want to _overwrite_ that text except for their handful of `auto_pairs`?
- Markdowners could still amend their auto_pairs for that lexer or their whole Textadept and retain their additional-autopair-chars.

I mean, would you consider a PR on something like that, or is it a categorical "Textadept shouldn't move that way anyway"-ish ?  =)

**Comment by orbitalquark** [comment]
I am open to more concrete suggestions, but I currently like the default behavior and find it sufficient for my needs.

**Comment by amitbha** [comment]
Anothor way:
~~~lua
events.connect(events.LEXER_LOADED, function(name)
if name == 'markdown' then
    keys['ctrl+b'] = function() textadept.editing.enclose("**", "**", true) end
end
end)
~~~
